web/client/epics/playback.js

```diff
@@ -346,23 +349,47 @@ export const playbackCacheNextPreviousTimes = (action$, { getState = () => { } }
             const snapType = snapTypeSelector(getState());
-            return Rx.Observable.forkJoin(
-                // TODO: find out a way to optimize and do only one request
-                // TODO: support for local list of values (in case of missing multidim-extension)
-                getDomainValues(...domainArgs(getState, { sort: "asc", limit: 1, fromValue: time, ...(snapType === 'end' ? {fromEnd: true} : {}) }))
-                    .map(res => res.DomainValues.Domain.split(","))
-                    .map(([tt]) => tt).catch(err => err && Rx.Observable.of(null)),
-                getDomainValues(...domainArgs(getState, { sort: "desc", limit: 1, fromValue: time, ...(snapType === 'end' ? {fromEnd: true} : {}) }))
-                    .map(res => res.DomainValues.Domain.split(","))
-                    .map(([tt]) => tt).catch(err => err && Rx.Observable.of(null))
-            ).map(([next, previous]) =>
-                updateMetadata({
+            return getTimeDomainsObservable(domainArgs, false, getState, snapType, time).map(([next, previous]) => {
+                return updateMetadata({
                     forTime: time,
                     next,
                     previous
-                })
-            );
+                });
+            });
         });
+
+/**
+ * Get domains with a slight buffer to detect whether the layer consists of
+ * instants/point or intervals/bars time value. The results is used to
+ * disable/enable the radio buttons to snap to start/end of time interval
+ */
+export const setIsIntervalData = (action$, { getState = () => { } } = {}) =>
+    action$.ofType(SELECT_LAYER, SET_CURRENT_TIME)
+        .filter(({type, layerId}) => (type === SET_CURRENT_TIME || (type === SELECT_LAYER && layerId)))
+        .switchMap(({time: actionTime}) => {
+            const time = actionTime || currentTimeSelector(getState());
+            const snapType = snapTypeSelector(getState());
+            return getTimeDomainsObservable(domainArgs, true, getState, snapType, time)
+                .map(([next, previous]) => {
+                    const isTimeIntervalData = next.indexOf('/') !== -1 || previous.indexOf('/') !== -1;
+                    return setIntervalData(isTimeIntervalData);
+                });
+        });
+
+/**
+ * In case a layer in the timeline is unselected from the TOC the timeline
+ * settings for snapping for layers are toggled off, this is to avoid persistence
+ * of selection of a non-visible layer on the timeline state, causing inconsistencies
+ * in case of mixed (point/interval) time based layers
+ */
+export const switchOffSnapToLayer = (action$, { getState = () => { } } = {}) =>
+    action$.ofType(CHANGE_LAYER_PROPERTIES)
+        .filter(({newProperties, layer}) => {
+            const selectedLayer = selectedLayerSelector(getState());
+            return (newProperties?.visibility !== undefined && !newProperties.visibility && selectedLayer === layer);
```

Comment:
I think this condition should be enough
```suggestion
            return ( !newProperties.visibility && selectedLayer === layer);
```

please apply it